deploy/runtools/run_farm.py

```diff
@@ -102,63 +118,63 @@ def allocate_host_port(self):
         return retport
 
 class F1_Instance(EC2Inst):
-    FPGA_SLOTS = 0
+    FPGA_SLOTS: int = 0
+    fpga_slots: List[FireSimServerNode]
+    fpga_slots_consumed: int
 
-    def __init__(self):
+    def __init__(self) -> None:
+        super().__init__()
         self.fpga_slots = []
         self.fpga_slots_consumed = 0
-        super(F1_Instance, self).__init__()
 
-    def get_num_fpga_slots_max(self):
+    def get_num_fpga_slots_max(self) -> int:
         """ Get the number of fpga slots. """
         return self.FPGA_SLOTS
 
-    def get_num_fpga_slots_consumed(self):
+    def get_num_fpga_slots_consumed(self) -> int:
         """ Get the number of fpga slots. """
         return self.fpga_slots_consumed
 
-    def add_simulation(self, firesimservernode):
+    def add_simulation(self, firesimservernode: FireSimServerNode) -> None:
```

Comment:
I reworked the `add_simulation` and `add_switch` to avoid having to use `Optional` types. This is still the same functionality as before.